src/python/pants/java/distribution/distribution.py

```diff
@@ -154,7 +154,7 @@ def home(self):
         if self._is_executable(os.path.join(jdk_dir, 'bin', 'javac')):
           home = jdk_dir
       self._home = home
-    return self._home
+    return text_type(self._home)
```

Comment:
Ideally we should have this guarantee when we construct the object... What codepaths fail if you remove this?

cc @Eric-Arellano